### PR TITLE
fix: support `SharedArrayBuffer` in `HttpResponse.arrayBuffer`

### LINKED ISSUE #2388: Using Typescript lib es2024 and arrayBuffer, types are hard to use
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 18 or higher

### Node.js version

v22.12.0

### Reproduction repository

https://github.com/danilofuchs/examples/blob/main/index.ts

### Reproduction steps

- Set lib to `es2024` or `esnext` to use features such as `Object.groupBy`
- Read a file using `readFile` or create a mocked buffer using `Buffer.from("fake-file")`
- Try to pass this to `HttpResponse.arrayBuffer`

### Current behavior

Typescript does not allow, but msw accepts it in runtime:

image

### Expected behavior

Previous lib versions works as expected

This is due to the introduction of `SharedArrayBuffer` in lib es2024

Using the output of readFileSync is documented as supported by msw:
image

### PR DESCRIPTION
**Pull request by danilofuchs**
Fixes https://github.com/mswjs/msw/issues/2388

I've added support for using `SharedArrayBuffer` as an argument to `HttpResponse.arrayBuffer()`

This is necessary as in Typescript 5.7 with lib ES2024, `Buffer` is no longer assignable to simply `ArrayBuffer`

When testing this, the default content type was being `text/plain;charset=utf-8`

I'm not sure if I did the right thing by setting the default to `application/octet-stream` (this is the only method without a default)

### CODE REVIEW

File: src/core/HttpResponse.test.ts [hunk 1]
```diff
@@ -253,18 +253,89 @@ describe('HttpResponse.html()', () => {
   })
 })
 
-it('creates an array buffer response', async () => {
-  const buffer = new TextEncoder().encode('hello world')
-  const response = HttpResponse.arrayBuffer(buffer)
+describe('HttpResponse.arrayBuffer()', () => {
```
Comment (kettanaito) [inline src/core/HttpResponse.test.ts:256]: Let's drop this `describe()` and adjust the test case names accordingly.

**Review by kettanaito — APPROVED** [review]
This looks great! Thank you so much for fixing this, @danilofuchs 👏

**Comment by kettanaito** [comment]
The Node.js tests are failing to unrelated reasons (see #2386). I'd love to fix that one first before merging this one so we get clean test runs.

**Comment by kettanaito** [comment]
Thanks once again for shipping this, @danilofuchs. Welcome to contributors! 🎉

**Comment by kettanaito** [comment]
## Released: v2.6.9 🎉

This has been released in v2.6.9!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v2.6.9)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/2.6.9)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Comment by danilofuchs** [comment]
Happy to help!

**Pull request merged**